### ISSUE TITLE
Single-letter gene "T" is not fully supported

### ISSUE THREAD

**Issue by ifokkema** [LOVDnl/LOVD3#106]
The single-letter gene "T" is not fully supported. LOVD does not like DBIDs in the format `T_000001`. Mutalyzer services seem to work fine.
See also: https://github.com/LOVDnl/geneloader/issues/4

**Comment by mkroon1** [comment]
The regex for checking the DBID format seems overly complex compared to the [HGNC specs](http://www.genenames.org/about/guidelines#genesymbols).

I've checked the [list of HGNC symbols](ftp://ftp.ebi.ac.uk/pub/databases/genenames/new/tsv/hgnc_complete_set.txt) to see if other genes give problems too:

~~~
$ grep -n -v -P '^(chr(\d{1,2}|[XYM])|(C(\d{1,2}|[XYM])orf[\d][\dA-Z]*-|[A-Z][A-Z0-9]+-)?(C(\d{1,2}|[XYM])orf[\d][\dA-Z]*|[A-Z][A-Z0-9-]+))$'
1243:APOBEC3A_B
2821:C4B_2
3632:CCD#
6763:DNM1DN1@
6764:DNM1DN2@
7257:ED#
8786:FAUNA@
9222:FOX#
10096:GLC#
10735:GTF2H2C_2
11809:HOXA@
11824:HOXB@
11838:HOXC@
11849:HOXD@
12273:IFN1@
12419:IGHD@
12420:IGHDOR15@
12439:IGHJ@
12477:IGHV2@
12532:IGHV3@
12570:IGHV@
12613:IGHVOR15@
12614:IGHVOR16@
12625:IGKJ@
12734:IGKV@
12735:IGKVOR2@
12736:IGKVP@
12745:IGLC@
12755:IGLJ@
12823:IGLV@
22174:OFC#
22228:OLFR@
23635:PCDHA@
23658:PCDHB@
23659:PCDHG@
23986:PGA@
25139:PR@
26785:RN5S1@
26786:RN5S2@
26787:RN5S3@
33043:SAA@
33168:SCAF#
34476:SMA#
34894:SNORD3@
35088:SNORD113@
35120:SNORD114@
35169:SNORD115@
35200:SNORD116@
36410:T
36747:TCRBVO@
38148:TRBVOR9@
39665:UGT2B@
~~~

The minimal change to the regex I came up with to allow these: `^(chr(\d{1,2}|[XYM])|(C(\d{1,2}|[XYM])orf[\d][\dA-Z]*-|[A-Z][A-Z0-9]+-)?(C(\d{1,2}|[XYM])orf[\d][\dA-Z]*|[A-Z][A-Z0-9-_#@]*))$`. Is implemented in 48467b0.

**Comment by mkroon1** [comment]
Fix ready in #194. Although note:

> Was not able to test all previously disallowed gene symbols. Because creation of `GTF2H2C_2` and `C4B_2` seemed to fail at mutalyzer's side, which may be a separate issue.

**Comment by ifokkema** [comment]
Hi Mark!
Do any of these "genes" really work? I was under the impression that none of these function, or make sense as a gene in LOVD. For instance, the `PCDHA@` type "genes" are gene clusters and don't have transcripts of their own, and us such they're not useful for us. Have we tested all of these?

**Comment by mkroon1** [comment]
No I haven't tested them all. HGNC denotes `PCDHA@` as "gene with protein product" though, but I can't find an associated transcript. `GTF2H2C_2` does have a transcript.

Isn't it easier and more consistent to just allow all valid HGNC symbols, instead of checking all of them for having transcripts?

**Comment by ifokkema** [comment]
> No I haven't tested them all. HGNC denotes PCDHA@ as "gene with protein product" though, but I can't find an associated transcript.

The HGNC should create a new type though, since this is a gene cluster. And as a cluster, they apparently decided to create a symbol for that, replacing the possible numbers with an `@`. So the gene cluster itself is not useful for us, as it doesn't have a transcript. Variants can just be described on all the genes individually. LOVD should therefore ignore (nicely deny creation of) all gene clusters.

> GTF2H2C_2 does have a transcript.

But no genomic location, so also this gene is not useful for us.

> Isn't it easier and more consistent to just allow all valid HGNC symbols, instead of checking all of them for having transcripts?

If we know, by looking at the name already, that they'll fail, we should better give people a nice warning instead of a fatal Mutalyzer error, which will happen otherwise. When I wrote this test, I checked all genes that didn't match (but apparently, I missed `T` or it didn't exist yet or so), and concluded that none of them worked. I don't mind how we do it, but these genes are not supported and the user should get a nice message about this, other than the fatal Mutalyzer error.
